src/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -178,6 +179,15 @@ public SAMFileWriterFactory setTempDirectory(final File tmpDir) {
     }
 
     /**
+     * Set the flag output format only when writing text.
+     * Default value: [[htsjdk.samtools.SAMTextWriter.samFlagFieldOutput.DEFAULT]]
+     */
+    public SAMFileWriterFactory setSamFlagFieldOutput(final SamFlagField samFlagFieldOutput) {
```

Comment:
I think passing null here will result in a blow up later on. If that's true then check non null here and blow up early. Otherwise, if null is OK, then document that null is OK.